File: smartadapter/src/main/java/smartadapter/internal/utils/ViewEventValidator.java
package smartadapter.internal.utils;

/*
 * Created by Manne Öhlund on 2019-06-11.
 * Copyright (c) All rights reserved.
 */

import java.util.Arrays;
import java.util.List;

import io.github.manneohlund.smartrecycleradapter.R;

/**
 * Checks if auto view events are of the predefined types provided by the SmartRecyclerAdapter library.
 */
public class ViewEventValidator {

    private static final List autoViewEvents = Arrays.asList(
            R.id.undefined,
            R.id.event_on_click,
            R.id.event_on_long_click,
            R.id.event_on_item_selected
    );

    public static boolean isViewEventIdValid(int viewEventId) {
        return autoViewEvents.contains(viewEventId);
    }
}

File: smartadapter/src/main/java/smartadapter/internal/utils/ViewEventValidator.kt
package smartadapter.internal.utils

/*
 * Created by Manne Öhlund on 2019-06-11.
 * Copyright (c) All rights reserved.
 */

import smartadapter.listener.OnItemClickListener
import smartadapter.listener.OnItemLongClickListener
import smartadapter.listener.OnViewEventListener

/**
 * Checks if auto view events are of the predefined types provided by the SmartRecyclerAdapter library.
 */
@Deprecated("Use custom view event binder")
object ViewEventValidator {

    private val autoViewEvents = listOf(
        OnItemClickListener::class,
        OnItemLongClickListener::class
    )

    fun isViewEventIdValid(viewEventId: OnViewEventListener<*>): Boolean {
        return autoViewEvents.contains(viewEventId::class)
    }
}
